src/Microsoft.AspNetCore.Server.IISIntegration/AuthenticationHandler.cs

```diff
@@ -1,138 +1,99 @@
-// Copyright (c) .NET Foundation. All rights reserved.
+﻿// Copyright (c) .NET Foundation. All rights reserved.
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
 using System;
-using System.Linq;
-using System.Security.Claims;
+using System.Globalization;
+using System.Security.Principal;
 using System.Threading.Tasks;
-using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Authentication;
 using Microsoft.AspNetCore.Http;
-using Microsoft.AspNetCore.Http.Authentication;
-using Microsoft.AspNetCore.Http.Features.Authentication;
 using Microsoft.Extensions.Internal;
+using Microsoft.Extensions.Primitives;
 
 namespace Microsoft.AspNetCore.Server.IISIntegration
 {
     internal class AuthenticationHandler : IAuthenticationHandler
     {
-        internal AuthenticationHandler(HttpContext httpContext, IISOptions options, ClaimsPrincipal user)
-        {
-            HttpContext = httpContext;
-            User = user;
-            Options = options;
-        }
-
-        internal HttpContext HttpContext { get; }
-
-        internal IISOptions Options { get; }
-
-        internal ClaimsPrincipal User { get; }
+        private const string MSAspNetCoreWinAuthToken = "MS-ASPNETCORE-WINAUTHTOKEN";
 
-        internal IAuthenticationHandler PriorHandler { get; set; }
+        internal AuthenticationScheme Scheme { get; private set; }
 
-        public Task AuthenticateAsync(AuthenticateContext context)
+        public Task<AuthenticateResult> AuthenticateAsync(AuthenticateContext context) 
         {
-            if (ShouldHandleScheme(context.AuthenticationScheme))
+            var user = GetUser(context.HttpContext);
+            if (user != null)
             {
-                if (User != null)
-                {
-                    context.Authenticated(User, properties: null, description: null);
-                }
-                else
-                {
-                    context.NotAuthenticated();
-                }
+                return Task.FromResult(AuthenticateResult.Success(new AuthenticationTicket(user, Scheme.Name)));
```

Comment:
Yes, but those scenarios don't usually overlap. E.g. you don't do NTLM + Cookies at the same time. So long as we can make Merge smart enough not to merge when there's really just WindowsPrincipal and Anonymous it should be fine.